src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,43 +1,84 @@
-import React from 'react';
+import React, { Dispatch, SetStateAction, useEffect, useState } from 'react';
+import cn from 'classnames';
+import { Todo } from '../../types/Todo';
+import { User } from '../../types/User';
+import { getUser } from '../../api';
 import { Loader } from '../Loader';
 
-export const TodoModal: React.FC = () => {
+type Props = {
+  selectedTodo: Todo | null;
+  setSelectedTodo: Dispatch<SetStateAction<Todo | null>>;
+};
+
+export const TodoModal: React.FC<Props> = ({
+  selectedTodo,
+  setSelectedTodo,
+}) => {
+  const [user, setUser] = useState<User | null>(null);
+  const [isLoadingModal, setIsLoadingModal] = useState(false);
+
+  useEffect(() => {
+    if (selectedTodo) {
+      setIsLoadingModal(true);
+      getUser(selectedTodo.userId).then(userData => {
+        setUser(userData);
+        setIsLoadingModal(false);
+      });
+    }
+  }, [selectedTodo]);
+
+  function handleCloseTodo() {
+    setSelectedTodo(null);
+  }
+
   return (
-    <div className="modal is-active" data-cy="modal">
-      <div className="modal-background" />
-
-      {true ? (
-        <Loader />
-      ) : (
-        <div className="modal-card">
-          <header className="modal-card-head">
-            <div
-              className="modal-card-title has-text-weight-medium"
-              data-cy="modal-header"
-            >
-              Todo #2
-            </div>
+    <>
+      <div className="modal is-active" data-cy="modal">
+        <div className="modal-background" />
+        {isLoadingModal ? (
+          <Loader />
+        ) : (
+          <div className="modal-card">
+            <header className="modal-card-head">
+              <div
+                className="modal-card-title has-text-weight-medium"
+                data-cy="modal-header"
+              >
+                Todo #{selectedTodo?.id}
+              </div>
 
-            {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-            <button type="button" className="delete" data-cy="modal-close" />
-          </header>
+              <button
+                type="button"
+                className="delete"
+                data-cy="modal-close"
+                aria-label="Close modal window"
+                onClick={() => handleCloseTodo()}
```

Comment:
```suggestion
                onClick={handleCloseTodo}
```

can be a bit simplify